plugins/isplink-client-registration/src/public.php

```diff
@@ -0,0 +1,229 @@
+<?php
+require_once __DIR__ . '/vendor/autoload.php';
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+$configManager = \Ubnt\UcrmPluginSdk\Service\PluginConfigManager::create();
+$config = $configManager->loadConfig();
+$optionsManager = \Ubnt\UcrmPluginSdk\Service\UcrmOptionsManager::create();
+$options = $optionsManager->loadOptions();
+
+$apiError = null;
+$apiSuccess = false;
+
+$publicUrl = str_replace('.php', '/', $options->pluginPublicUrl);
+
+
+try {
+    $client = new Client([
+        'base_uri' => "api.isplink.net",
+        'headers' => [
+            'Authorization' => 'Bearer ' . $config['API_KEY'],
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json'
+        ],
+        'timeout' => 5,
+        'http_errors' => true 
+    ]);
+
+    // Test API connection
+    $response = $client->post('/plugin/verify');
+    $apiSuccess = true;
+    $body = (string) $response->getBody();
+    $jsonData = json_decode($body, true);
+    $clientDomain = $jsonData['client_domain'];
+    
+} catch (GuzzleHttp\Exception\ClientException $e) {
+    // Handle 4xx client errors (like 401 Unauthorized)
+    $response = $e->getResponse();
+    $body = (string) $response->getBody();
+    $jsonData = json_decode($body, true);
+    $clientDomain = false;
+    
+    // ISP link is hyperlink to ISP Link website
+    if (isset($jsonData['error']) && $jsonData['error'] === 'Invalid API Key') {
+        $apiError = "An API key and active ISPLink account is required for this plugin. Please contact <a href='https://isplink.app'>ISPLink</a> to activate your account.";
+    } else {
+        $apiError = 'Client error: ' . $response->getStatusCode() . ' - ' . ($jsonData['error'] ?? 'Unknown error');
+    }
+    error_log('API error: ' . $apiError);
```

Comment:
Could you consider logging into the plugin log file?

Reply:
Logging should now be using the Plugin logging SDK.